File: LeetcodeQuestions/leetcode_challenges/February_challange/validate_stack_sequence.py
def validateStackSequences(pushed, popped):
    i = 0
    j = 0
    stack = []
    while j < len(popped):
        if len(stack) and popped[j] == stack[-1]:
            stack.pop()
            j += 1
        elif i < len(pushed):
            stack.append(pushed[i])
            i += 1
        else:
            return False
    return True


print(validateStackSequences([1, 2, 3, 4, 5], [4, 5, 3, 2, 1]))
print(validateStackSequences([1, 2, 3, 4, 5], [4, 3, 5, 1, 2]))

File: LeetcodeQuestions/leetcode_challenges/February_challange/validate_stack_sequence.rb
def validate_stack_sequences(pushed, popped)
    i = 0 
    j = 0
    stack = []
    
    while j < popped.length
        if popped[j] == stack[stack.length - 1]
            stack.pop()
            j += 1
        elsif i < pushed.length
            stack.push(pushed[i])
            i += 1
        else
            return false
        end
    end
    return true
end
